rio_tiler/models.py

```diff
@@ -787,3 +790,43 @@ def statistics(
             f"{self.band_names[ix]}": BandStatistics(**stats[ix])
             for ix in range(len(stats))
         }
+
+    def get_coverage_array(
+        self,
+        shape: Dict,
+        shape_crs: CRS = WGS84_CRS,
+        cover_scale: int = 10,
+    ) -> numpy.array:
+
+        """
+        cover_scale: int, optional
+            Scale used when generating coverage estimates of each
+            raster cell by vector feature. Coverage is generated by
+            rasterizing the feature at a finer resolution than the raster then using a summation to aggregate
+            to the raster resolution and dividing by the square of cover_scale
+            to get coverage value for each cell. Increasing cover_scale
+            will increase the accuracy of coverage values; three orders
+            magnitude finer resolution (cover_scale=1000) is usually enough to
+            get coverage estimates with <1% error in individual edge cells coverage
+            estimates, though much smaller values (e.g., cover_scale=10) are often
+            sufficient (<10% error) and require less memory.
+
+        Note: code adapted from https://github.com/perrygeo/python-rasterstats/pull/136 by @sgoodm
+
+        """
+        if self.crs != shape_crs:
+            shape = transform_geom(shape_crs, self.crs, shape)
+
+        cover_array = rasterize(
+            [(shape, 1)],
+            out_shape=(self.height * cover_scale, self.width * cover_scale),
+            transform=self.transform * Affine.scale(1 / cover_scale),
+            all_touched=True,
+            fill=0,
+            dtype="uint8",
+        )
+        cover_array = cover_array.reshape(
+            (self.height, cover_scale, self.width, cover_scale)
+        ).astype("float32")
+
+        return cover_array.sum(-1).sum(1) / (cover_scale**2)
```

Comment:
slightly modified version of https://github.com/perrygeo/python-rasterstats/pull/136

@sgoodm I know ☝️ is a 7 years old PR (😅) but I hope you don't mind that I reused some of the code here 🙏

Reply:
Not at all, @jacobwhall and I have been pushing use of COGs in our work at @aiddata and are fans of these projects, so happy to share some code to support :+1: :+1: